app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb

```diff
@@ -0,0 +1,515 @@
+class ManageIQ::Providers::Redhat::Inventory::Parser::InfraManager < ManageIQ::Providers::Redhat::Inventory::Parser
+  def parse
+    log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS name: [#{collector.manager.name}] id: [#{collector.manager.id}]"
+    $rhevm_log.info("#{log_header}...")
+
+    datacenters
+    clusters
+    storagedomains
+    hosts
+    vms
+
+    $rhevm_log.info("#{log_header}...Complete")
+  end
+
+  def clusters
+    collector.clusters.each do |cluster|
+      persister.resource_pools.find_or_build(r_id).assign_attributes(
+        :name       => "Default for Cluster #{cluster.name}",
+        :uid_ems    => "#{cluster.id}_respool",
+        :is_default => true,
+      )
+
+      ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(cluster.href)
+
+      persister.clusters.find_or_build(cluster.id).assign_attributes(
+        :ems_ref       => ems_ref,
+        :ems_ref_obj   => ems_ref,
+        :uid_ems       => cluster.id,
+        :name          => cluster.name,
+        :datacenter_id => persister.datacenters.lazy_find(cluster.data_center.id).id,
+      )
+    end
+  end
+
+  def storagedomains
+    collector.storagedomains.each do |storagedomain|
+      storage_type = storagedomain.dig(:storage, :type).upcase
+      location = if storage_type == 'NFS' || storage_type == 'GLUSTERFS'
+                   "#{storagedomain.dig(:storage, :address)}:#{storagedomain.dig(:storage, :path)}"
+                 else
+                   logical_units = storagedomain.dig(:storage, :volume_group, :logical_units)
+                   logical_unit =  logical_units && logical_units.first
+                   logical_unit && logical_unit.id
+                 end
+
+      free        = storagedomain.try(:available).to_i
```

Comment:
you can probably put these  one liners directly to the hash below, to spare some LOC